screenpy/actor.py

```diff
@@ -99,13 +130,33 @@ def perform(self, action: Performable) -> None:
         """Perform an Action."""
         action.perform_as(self)
 
-    def cleans_up(self) -> None:
-        """Perform any scheduled clean-up tasks."""
+    def cleans_up_ordered_tasks(self) -> None:
+        """Perform ordered clean-up tasks."""
         try:
-            for task in self.cleanup_tasks:
+            for task in self.ordered_cleanup_tasks:
                 self.perform(task)
         finally:
-            self.cleanup_tasks = []
+            self.ordered_cleanup_tasks = []
+
+    def cleans_up_independent_tasks(self) -> None:
+        """Perform independent clean-up tasks."""
+        for task in self.independent_cleanup_tasks:
+            try:
+                self.perform(task)
+            except Exception as e:  # pylint: disable=broad-except
+                action = get_additive_description(task)
+                msg = (
+                    f"{self} encountered an error while attempting to {action}:"
+                    f"\n    {e}"
+                )
+                aside(msg)
+
+        self.independent_cleanup_tasks = []
+
+    def cleans_up(self) -> None:
+        """Perform any scheduled clean-up tasks."""
+        self.cleans_up_independent_tasks()
+        self.cleans_up_ordered_tasks()
```

Comment:
I'm trying to envision how this might break, e.g. if a user needs a mix of both but for some reason needs the ordered tasks to run first. 

But I can't think of a scenario where this logic either creates or exacerbates such a problem. 
If the user has that complex of a mix of cleanup tasks, they can certainly work around the issues by other means such as moving some of the tasks out of the cleanup logic and into their test as an informal pre-cleanup cleanup.

Reply:
Yeah, i put them in methods so the author would be able to call them in their teardown directly, if they need to.